### PR TITLE
feat: adds csp nonce support

### LINKED ISSUE #69: Add csp nonce support
See: https://github.com/styled-components/styled-components/issues/887

### PR DESCRIPTION
**Pull request by madou**
Closes #69 

TODO:
- [x] Add option getter inside babel plugin

### CODE REVIEW

File: examples/class-names-dynamic-object.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useState } from 'react';
 import { ClassNames } from '@compiled/css-in-js';
 
 export default {
-  title: 'class names dynamic object',
+  title: 'class names sdynamic object',
```
Comment (danieldelcore) [inline examples/class-names-dynamic-object.tsx:5]: typo

Comment (madou) [reply examples/class-names-dynamic-object.tsx:5]: gottem

**Comment by theKashey** [comment]
`nonce` is not supposed to be exposed at `window`. You should "hide" it, well, nobody explains how.
However `__webpack_nonce__` is not a global variable - it is a local variable, injected via _local module variable_.

I would propose use a module-in-the-middle to handle this stuff - https://github.com/thekashey/get-nonce
It makes debugging/mocking a bit easier as well.

**Comment by theKashey** [comment]
In short - idea behind `nonce` - don't let some _non-trusted-stuff_ like ads or browser plugins inject code you do not expect. 
As a result exposing `nonce` as a global variable, or expecting that it could be read from a global variable, is ruing a whole idea.

**Comment by madou** [comment]
hey thanks anton good point. i'll change the example to use something hardcoded for now.

when it comes to actual usage - it just sets a variable which webpack can then pick up and modify - that would work fine right? (so no changes to the implementation needed)

**Comment by theKashey** [comment]
It's not about webpack. It is:
- about `csp` *headers* your server is sending (that is not webpack)
- or about `csp` *meta* you added into the `<head>`(that is not webpack)
- and about the `nonce` generation (that is not webpack)
- and the ability for different pieces to read and use `nonce` if it's required
- actually - `script nonce` and `style nonce` could be different, and the second it not that required.

So that's not so simple :(

**Comment by madou** [comment]
thanks for the info!

for this library though i can't imagine needing to do anything more? other than ensuring there is good documentation around how to use nonce properly. and re-enforcing what you said in your last point

**Pull request merged**